tools/deleteorphanlogcommand.class.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+
+use Glpi\Console\AbstractCommand;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Helper\ProgressBar;
+
+
+
+class DeleteOrphanLogCommand extends AbstractCommand {
+
+   protected $requires_db_up_to_date = false;
+
+   protected function configure() {
+      parent::configure();
+
+      $this->setName('glpi:tools:delete_orphan_logs');
+      $this->setAliases(
+         [
+            'tools:delete_orphan_logs',
+         ]
+      );
+      $this->setDescription(__('Delete orphan log'));
+
+      $this->addOption(
+         'dry-run', '',
+         InputOption::VALUE_NONE,
+         __('Simulate the command without actually delete anything')
+
+      );
+
+   }
+
+   protected function execute(InputInterface $input, OutputInterface $output) {
+
+      $no_interaction = $input->getOption('no-interaction');
+      if (!$no_interaction) {
```

Comment:
Not sure it is necessary to ask for confirmation if dry-run mode is active.
```suggestion
      $no_interaction = $input->getOption('no-interaction');
      $dry_run = $input->getOption('dry-run')
      if (!$no_interaction && !$dry_run) {
```